src/runtime/fetcher.js

```diff
@@ -46,18 +54,44 @@ export class XhrFetcher {
     this.xhr_ = new Xhr(win);
   }
 
-  /** @override */
-  fetchCredentialedJson(url) {
-    const init = /** @type {!../utils/xhr.FetchInitDef} */ ({
-      method: 'GET',
+  /**
+   *
+   * @param {string=} method
+   * @return {!../utils/xhr.FetchInitDef}
+   */
+  getCredentialedInit_(method) {
+    return /** @type {!../utils/xhr.FetchInitDef} */ ({
+      method: method || 'GET',
       headers: {'Accept': 'text/plain, application/json'},
       credentials: 'include',
     });
-    return this.xhr_.fetch(url, init).then(response => response.json());
+  }
+
+  /** @override */
+  fetchCredentialedJson(url) {
+    return this.fetch(url, this.getCredentialedInit_()).then(response =>
+      response.json()
+    );
   }
 
   /** @override */
   fetch(url, init) {
     return this.xhr_.fetch(url, init);
   }
+
+  /** @override */
+  sendBeacon(url, data) {
+    const newUrl = addQueryParam(
```

Comment:
Could we just modify `url` here? That way later on in the method, there doesn't have to be the concept of a `newUrl`